UW433: Forecast docs (#389)
ufs-community/uwtools: docs/sections/user_guide/uw_yaml/field_table_yaml.rst

.. _defining_a_field_table:

Defining a ``field_table``
==========================

> The :ufs-weather-model:`UFS Weather Model<>` requires as one of its input files (:weather-model-io:`documented here<model-configuration-files>`) a ``field_table`` file. While the format required by the ufs-weather-model is not a YAML file, with ``uwtools`` package, the ``field_table`` entries may be managed as a YAML file, and written to the expected structure (:weather-model-io:`documented here<field-table-file>`).

Reviewer: ```suggestion
The :ufs-weather-model:`UFS Weather Model<>` requires as one of its inputs (:weather-model-io:`documented here<model-configuration-files>`) a ``field_table`` file in a :weather-model-io:`custom format<field-table-file>. Its content can be defined in UW YAML, then created with the ``uw`` CLI.
```

To generate a given ``field_table`` entry with the form:

.. code-block:: text

   "TRACER", "atmos_mod", "sphum"
             "longname",     "specific humidity"
             "units",        "kg/kg"
             "profile_type", "fixed", "surface_value=3.e-6" /

> Entries can be manage in YAML with the form:

Reviewer: ```suggestion
Entries can be represented in YAML with the form:
```

> Entries can be manage in YAML with the form:

Reviewer: Or `managed` if you like that better -- not `manage`, though.

.. code-block:: text

   sphum:

>      longname: "specific humidity"

Reviewer: Doesn't really matter, but FYI the quotes here are optional. YAML will parse it as the same string either way. The quotes are kind of voodoo in this case. I only bring it up because it may obscure an understanding of how YAML really works, and the power of using quotes when they're _really_ needed, i.e. to force a value that normally would be parsed as something else to be parsed as a string.

     units: kg/kg
     profile_type:
       name: fixed
       surface_value: 3.e-6

Additional tracers may be added at the same level of YAML indentation as ``sphum:`` in the above example.

UW YAML Keys
------------

``sphum:``
^^^^^^^^^^

> The name of the tracer. The entry corresponds to the name chosen for the first line of the native ``field_table`` entry.

Reviewer: ```suggestion
The short name of the tracer, to create the corresponding ``field_table`` entry's first line.
```

``longname:``
^^^^^^^^^^^^^

The descriptive name of the tracer.

``units:``
^^^^^^^^^^

> The units for the tracer. This entry directly corresponds to the tracer units in the native ``field_table`` entry.

Reviewer: ```suggestion
The units of the tracer, to create the corresponding ``field_table`` entry's ``"units"`` line.
```
